lib/magma/attribute.rb

```diff
@@ -116,57 +137,36 @@ def json_template
       }.delete_if {|k,v| v.nil? }
     end
 
-    def json_for record
-      record[ name ]
-    end
-
-    def txt_for(record)
-      json_for record
-    end
-
-    def update_record(record, new_value)
-      record.set({name=> new_value})
-
-      if database_type == DateTime
-        return DateTime.parse(new_value)
-      elsif database_type == Float
-        return new_value.to_f
-      elsif database_type == Integer
-        return new_value.to_i
-      else
-        return new_value
-      end
+    def query_to_payload(value)
+      value
     end
 
-    def read_only?
-      read_only
+    def query_to_tsv(value)
+      query_to_payload(value)
     end
 
-    def shown?
-      !hidden
+    def revision_to_loader(record_name, new_value)
+      [
+        name,
+        database_type == DateTime ?
```

Comment:
To reduce the ternary operator soup you can use
```
case database_type
when DateTime
   ...
when Integer
  ...
when Float
   ..

end
```

Reply:
Also, being able to convert a new value based on the attribute's database_type seems like a generally useful function to break out as separate function?

Reply:
I think that is the (only) purpose of this function, it bizarrely returns a [ name,  ... ] wrapped around the value => entry conversion because of the poor interface design/coupling with the corresponding usage of this method by the update endpoint to pass data between Revision and Loader, attribute-specifically. I wasn't really able to solve this problem very well, thus this ugly mess.

Reply:
It occurs to me that this was a result of the previous version of Magma::Attribute, which was overloaded with a bunch of different type behaviors. These have been separated out into various actual attribute types (e.g. IntegerAttribute) which each define their own database_type - this method should ideally be broken up into those attribute classes (with the default here just `[ name, new_value ]`)